mountpoint-s3/src/upload.rs

```diff
@@ -20,12 +20,16 @@ pub struct Uploader<Client> {
 #[derive(Debug)]
 struct UploaderInner<Client> {
     client: Arc<Client>,
+    storage_class: Option<String>,
 }
 
 impl<Client: ObjectClient> Uploader<Client> {
     /// Create a new [Uploader] that will make requests to the given client.
-    pub fn new(client: Arc<Client>) -> Self {
-        let inner = UploaderInner { client };
+    pub fn new(client: Arc<Client>, storage_class: Option<String>) -> Self {
```

Comment:
This one should still be `storage_class: String` -- our Uploader will always set the storage class, it's just the `mountpoint-s3-client` crate that shouldn't force a storage class to be specified.